soc/nxp/imxrt/Kconfig

```
@@ -42,21 +42,25 @@ config BOOT_XSPI_NOR
 	bool "XPI serial NOR"
 	depends on DT_HAS_NXP_XSPI_ENABLED
 
+FLEXSPI_COMPAT := nxp,imx-flexspi
+
 config BOOT_FLEXSPI_NOR
 	bool "FlexSPI serial NOR"
-	depends on HAS_MCUX_FLEXSPI
+	depends on $(dt_has_compat,$(FLEXSPI_COMPAT))
```

Shouldn't it be limited to enabled nodes?

```suggestion
	depends on $(dt_compat_enabled,$(FLEXSPI_COMPAT))
```

---

no, it's intentional, these config are used for boot configuration regardless of if flexspi driver is used